Wayne JobTemplate View

wbkirbs opened the pull request:
Updated views for job template list and job template detail.

Relative URLs:
/templates
/template_detail/2

Commit:
save

Commit:
Added views for job template list, and details of a single job template

jcaxmacher commented on tasker/job_template/views.py:
```diff
@@ -1,21 +1,16 @@
-    {'id' : 2, 'title':'Template 3', 'start_date' : '09.20.2020', 'interval' : 1, 'interval_type' : 'Months', 'description' : 'Job Template description goes here.'}
-    ]
-    return render_template('job-template/templates.html', title='Templates', user=user, templates=templates)
+    jobs = JobTemplate.getJobTemplates()
```
I think it would be simpler to do the query here than delegate to the JobTemplate class.  Also, the query needs to select only templates for the logged in user.

wbkirbs replied:
I agree with simpler, but just trying to keep the MVC paradigm.  I meant to add the user filter, but then it slipped my mind.  I'll add that shortly.

ekmartens replied:
In Flask applications, the route (which we have been putting in a file called views, which may not be particularly accurate) acts as the controller and the template is the view. While we _can_ define our own class functions, SQLAlchemy functions can be considered the model functions. So performing the query with SQLAlchemy functions in the route definition still upholds the MVC paradigm.

jcaxmacher commented on tasker/job_template/views.py:
```diff
@@ -25,16 +20,9 @@ def add_template():
-    ]
-    template = templates[int(id)]
-    return render_template('job-template/template-detail.html', title='Template Details', user=user, template=template)
+    job = JobTemplate.getJobTemplate(id)
```
Same comment as above on making sure the job template with the given ID belongs to the user.  This view also needs to `raise NotFound('Job template not found')` if the template with ID does not exist or it exists but belongs to different user.

ekmartens replied:
You can check that the job template owner is the current logged in user with current_user (you'll need to test it with a logged in test account):

from flask_login import current_user

Commit:
Moved query code to views.py
Modified queries to filter based on user id of logged in user
Added exception for job template not found

Commit:
Restored requirements.txt

jcaxmacher committed:
Merge branch 'master' into wayne-jobtemplate-view

jcaxmacher commented:
Looks good!
